src/pkg/middleware.ts

```diff
@@ -55,6 +55,16 @@ export const requireAuth =
   (createHelpers: RouteOptionsCreator) =>
   (req: Request, res: Response, next: NextFunction) => {
     const { frontend, apiBaseUrl } = createHelpers(req, res)
+
+    // when accessing settings with a valid flow id
+    // we allow the settings page to trigger the
+    // login flow on session_aal2_required
+    if (req.url.includes("/settings") && req.query.flow) {
+      if (isUUID.test(req.query.flow.toString())) {
```

Comment:
Why would this ever be false? If there is a flow parameter supplied, I'd just assume it's a valid, existing flow?

Reply:
Never know. Just trying to prevent as much redirects as possible. For example. You have the flow parameter but no uuid. Then instead of being redirected to settings and then back to kratos settings init and then again through whoami middleware and then to settings, it just checks the middleware immediately.